velox/exec/tests/HashPartitionFunctionTest.cpp

```diff
@@ -36,70 +36,96 @@ TEST_F(HashPartitionFunctionTest, HashPartitionFunction) {
 
     std::vector<uint32_t> partitonsWithBits(numRows);
     HashPartitionFunction functionWithBits(HashBitRange{0, 3}, rowType, {0});
     functionWithBits.partition(*vector, partitonsWithBits);
     EXPECT_NE(partitonsWithoutBits, partitonsWithBits);
+    EXPECT_EQ(8, functionWithBits.numPartitions());
   }
 
   // The test case the two hash partition functions with different configs.
   {
     std::vector<uint32_t> partitonsWithoutBits(numRows);
     HashPartitionFunction functionWithoutBits(4, rowType, {0});
     functionWithoutBits.partition(*vector, partitonsWithoutBits);
+    EXPECT_EQ(4, functionWithoutBits.numPartitions());
 
     std::vector<uint32_t> partitonsWithBits(numRows);
     HashPartitionFunction functionWithBits(HashBitRange{29, 31}, rowType, {0});
     functionWithBits.partition(*vector, partitonsWithBits);
     EXPECT_NE(partitonsWithoutBits, partitonsWithBits);
+    EXPECT_EQ(4, functionWithBits.numPartitions());
   }
 
   // The test case the two hash partition functions with different configs.
   {
     std::vector<uint32_t> partitonsWithBits1(numRows);
     HashPartitionFunction functionWithBits1(HashBitRange{40, 42}, rowType, {0});
     functionWithBits1.partition(*vector, partitonsWithBits1);
+    EXPECT_EQ(4, functionWithBits1.numPartitions());
 
     std::vector<uint32_t> partitonsWithBits2(numRows);
     HashPartitionFunction functionWithBits2(HashBitRange{29, 31}, rowType, {0});
     functionWithBits2.partition(*vector, partitonsWithBits2);
     EXPECT_NE(partitonsWithBits1, partitonsWithBits2);
+    EXPECT_EQ(4, functionWithBits2.numPartitions());
   }
 
   // The test case the two hash partition functions with different configs.
   {
     std::vector<uint32_t> partitonsWithBits1(numRows);
     HashPartitionFunction functionWithBits1(HashBitRange{20, 31}, rowType, {0});
     functionWithBits1.partition(*vector, partitonsWithBits1);
+    EXPECT_EQ(1 << 11, functionWithBits1.numPartitions());
 
     std::vector<uint32_t> partitonsWithBits2(numRows);
     HashPartitionFunction functionWithBits2(HashBitRange{29, 31}, rowType, {0});
     functionWithBits2.partition(*vector, partitonsWithBits2);
     EXPECT_NE(partitonsWithBits1, partitonsWithBits2);
+    EXPECT_EQ(4, functionWithBits2.numPartitions());
   }
 
   // The test case the two hash partition functions having the same config.
   {
     std::vector<uint32_t> partitonsWithBits1(numRows);
     HashPartitionFunction functionWithBits1(HashBitRange{29, 31}, rowType, {0});
     functionWithBits1.partition(*vector, partitonsWithBits1);
+    EXPECT_EQ(4, functionWithBits1.numPartitions());
 
     std::vector<uint32_t> partitonsWithBits2(numRows);
     HashPartitionFunction functionWithBits2(HashBitRange{29, 31}, rowType, {0});
     functionWithBits2.partition(*vector, partitonsWithBits2);
     EXPECT_EQ(partitonsWithBits1, partitonsWithBits2);
+    EXPECT_EQ(4, functionWithBits2.numPartitions());
   }
 }
+
+TEST_F(HashPartitionFunctionTest, HashBitRange) {
```

Comment:
Thank you for adding a test.

naming: HashBitRange -> hashBitRange

Also, it would be nice to move this test into its own file: HashBitRangeTest.cpp . (I'd move HashBitRange into its own file as well.)